src/plugins/data_table/data_table.js

```diff
@@ -2,30 +2,167 @@ var LocalDataTable = (function() {
 
   var Base = Backdraft.plugin("Base");
 
+  var DEFAULT_JST_DELIMS = {
+    evaluate    : /<%([\s\S]+?)%>/g,
+    interpolate : /<%=([\s\S]+?)%>/g,
+    escape      : /<%-([\s\S]+?)%>/g
+  };
+
+  var DataTableFilter = Base.View.extend({
+    filterTemplate : _.template('\
+      <div class="toggle-filter-button" data-toggle="dropdown">\
+        <span class="<%= filterButtonClass %>"></span>\
+      </div>\
+    ', null, DEFAULT_JST_DELIMS),
+
+    filterMenuTemplate : _.template('\
+      <div class="filterMenu dropdown-menu <%= listClass %>">\
+        <% if (filter.type === "string") { %>\
+          <input class="filter-string" id ="value" type="text" placeholder="Search <%= title %>" />\
+        <% } else if (filter.type === "numeric") { %>\
+          <ul>\
+            <li> &gt; <input id="gt" class="filter-numeric" type="text" /></li> \
+            <li> &lt; <input id="lt" class="filter-numeric" type="text"/></li> \
+            <li> = <input id="eq" class="filter-numeric" type="text" /></li> \
+          </ul>\
+        <% } else if (filter.type === "list") { %>\
+          <ul>\
+            <% _.each(filter.options, function(element, index) { %>\
+              <li>\
+                <label>\
+                  <input class="list" id="value" type="checkbox" name="<%= attr %>" value="<%= element %>" /> \
+                  <%= element %>\
+                </label>\
+              </li>\
+            <% }) %>\
+          </ul>\
+        <% } %>\
+        <button class="btn btn-primary btn-sm btn-filter" name="button" type="submit" title="">Filter</button>\
+        <button class="btn btn-primary btn-sm btn-clear" name="button" type="submit" title="">Clear</button>\
+      </div>\
+    ', null, DEFAULT_JST_DELIMS),
+
+    events: {
+      "click .toggle-filter-button": "_onToggleClick",
+      "click input": "_onInputClick",
+      "change input": "_onInputChange",
+      "click .btn-filter": "_onFilterClick",
+      "click .btn-clear": "_onClearClick"
+    },
+
+    initialize: function(options) {
+      this.filter = options.column.filter;
+      this.attr = options.column.attr;
+      this.title = options.column.title;
+      this.table = options.table;
+      this.head = options.head;
+      this.filterButtonClass = "filterInactive";
+    },
+
+    render: function() {
+      this.$el.html( this.filterTemplate( { filterButtonClass : this.filterButtonClass } ));
+      // build filter menu
+      this._buildFilterMenu();
+      return this;
+    },
+
+    _buildFilterMenu: function() {
+      // handle listClass
+      var listClass = "";
+      if (this.filter.type === "list") {
+        if (this.filter.options.length > 30) {
+          listClass = " triple";
+        } else if (this.filter.options.length > 15) {
+          listClass = " double";
+        } else {
+          listClass = " single";
+        }
+      }
```

Comment:
This can be integrated into the template even with the current template, it's just ugly.